### ISSUE TITLE
Support for remote client (feature request)

### ISSUE THREAD

**Issue by MatanZ** [transmission/transmission#5855]
### What is the issue?

Torrent clients usually allow the user to preview (or view) files within downloading/downloaded torrents. When the server and the the client run on the same system (as in the case when they are actually the same program), this is quite easy - the client knows the file name, checks the possible locations for the file, and previews it (or passes the file to a program according to the file type).

When the client connects to a remote daemon this is harder. The client needs a way to access the remote file system, and translating file name to remote file location may be trickier.

My suggestion is to allow passing of the file data from daemon over RPC. This can be done using a new RPC method (`file-get`), or by adding a parameter to the torrent-get call that will include the file data in the returned `files` array. In the latter case, I believe it is reasonable to limit to one file and and allowing the client to select a byte range.

### Which application of Transmission?

transmission-daemon

**Comment by killemov** [comment]
This would make transmission-daemon a full-fledged web-server. And that is what web-servers are for. I'm currently working on "content magic" for [Shift](https://github.com/killemov/Shift) making some content directly available in the browser. It starts with the "fileLink" attribute to give files a path to be served from a webserver, in my case [Lighty](https://www.lighttpd.net/).

**Comment by MatanZ** [comment]
Yes. it is possible to implement this feature with remote access to the server's file system (which need not be http. It can be ftp, ssh, ...) But this adds complications and chances for mistakes:
- `transmission-daemon` is already configured with password, IP allow/deny lists, firewall, other access controls. Adding another daemon for another protocol requires configuring all those things.
- There are a few locations possible for files (torrent location, incomplete torrent location). The way a lot of programs work requires exposing the lowest point in the tree common to both, which might expose a lot of the file system.
- The files may be in the same directory with non-torrent files. `transmission-daemon` only knows about the torrent files, but a file system exposing daemon will need to be specifically configured.
- As you noted, the daemon needs to be modified, anyway.
